scripts/profiler/model_creator.py

```diff
@@ -0,0 +1,298 @@
+        self.config = config
+        self.event_filename = event_filename
+        self.event_types_filename = event_types_filename
+        self.plotting = plotting
+        self.finish_event = finish_event
+        self.sock_desc = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+        self.sock_desc.bind(SOCK_RECV_DESC)
+        self.sock_ev = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+        self.sock_ev.bind(SOCK_RECV_EV)
+        if self.plotting:
+            self.sock_proc_ev = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+            self.sock_proc_ev.bind(SOCK_SEND_PROC_EV)
+        self.timestamp_overflows = 0
+        self.after_half = False
+
+        self.processed_events = ProcessedEvents()
+        self.temp_events = []
+        self.submitted_event_type = None
+
+        self.desc_buf = ""
+        self.bufs = list()
+        self.bcnt = 0
+        self.last_read_time = time.time()
+        self.reading_data = True
+
+        self.logger = logging.getLogger('Profiler model creator')
+        self.logger_console = logging.StreamHandler()
+        self.logger.setLevel(log_lvl)
+        self.log_format = logging.Formatter('[%(levelname)s] %(name)s: %(message)s')
+        self.logger_console.setFormatter(self.log_format)
+        self.logger.addHandler(self.logger_console)
+
+    def shutdown(self):
+        if self.event_filename and self.event_types_filename:
+            self.processed_events.write_data_to_files(self.event_filename,
+                                                      self.event_types_filename)
+
+    def _get_buffered_data(self, num_bytes):
+        buf = bytearray()
+        while len(buf) < num_bytes:
+            tbuf = self.bufs[0]
+            size = num_bytes - len(buf)
+            if len(tbuf) <= size:
+                buf = buf + tbuf
+                del self.bufs[0]
+            else:
+                buf = buf + tbuf[0:size]
+                self.bufs[0] = tbuf[size:]
+        self.bcnt -= num_bytes
+        return buf
+
+    def _read_bytes(self, num_bytes):
+        while self.reading_data:
+            if self.bcnt >= num_bytes:
+                break
+            bytes_address_pair = self.sock_ev.recvfrom(4096*4)
+            buf = bytes_address_pair[0]
+
+            if len(buf) > 0:
+                self.bufs.append(buf)
+                self.bcnt += len(buf)
+
+            if self.bcnt >= num_bytes:
+                break
+
+            time.sleep(0.05)
+
+        return self._get_buffered_data(num_bytes)
+
+    def _calculate_timestamp_from_clock_ticks(self, clock_ticks):
+        return self.config['ms_per_timestamp_tick'] * (
+            clock_ticks + self.timestamp_overflows * self.config['timestamp_raw_max']) / 1000
+
+    def transmit_all_events_descriptions(self):
+        bytes_address_pair = self.sock_desc.recvfrom(4096)
+        self.desc_buf = bytes_address_pair[0].decode()
+        while True:
+            desc = str(self.desc_buf[0:self.desc_buf.find('\n')])
```

Comment:
If this is transmitted as something known to python, let's use python json processing instead of developing our own

e.g.
`csv.reader`